packages/component/src/hooks/internal/useAcknowledgedActivity.js

```diff
@@ -0,0 +1,61 @@
+import { hooks } from 'botframework-webchat-api';
+import { useMemo, useRef } from 'react';
+import { useSticky } from 'react-scroll-to-bottom';
+
+import findLastIndex from '../../Utils/findLastIndex';
+import getActivityUniqueId from '../../Utils/getActivityUniqueId';
+import useChanged from './useChanged';
+
+const { useActivities } = hooks;
+
+// Acknowledged means either:
+// 1. The user sent a message
+//    - We don't need a condition here. When Web Chat send the message, it will scroll to bottom, and it will trigger condition 2 below.
+// 2. At the bottom of the transcript, scrolled from a non-bottom scroll position
```

Comment:
I'm a little confused by this section / the intent. 

This line seems to address to different scenarios. 
Bottom of the transcript = acknowledged. 

"scrolled from a non-bottom scroll position" - I'm assuming this refers to manual scrolling by the user, is that correct?

"Activity is acknowledged when user is already at the bottom of the transcript or user manually scrolls down to the bottom of the transcript" -- does this better summarize this line?

Reply:
"scrolled from a non-bottom scroll position", yes, this is manual scrolling by the user. I have updated it to "The user scroll to the bottom of the transcript, from a non-bottom scroll position".

Your statement is not exactly correct because of this part "activity is acknowledged when user is at the bottom of the transcript".

Consider this, the developer set "pause after 3 activities". And Web Chat received 1 single line activity, and another after 5 seconds (proactive), then nothing else. Sum it up: we received 2 activities, they are short, and the transcript is still sticky.

These 2 activities should not be acknowledged even though the user is at the bottom of the transcript.

If we acknowledged them, assume 2 more single-liner come in (total 4). Because the first 2 were already acknowledged, scroll will not pause when last 2 come in. Then we violated "pause after 3 activities".

I rewrote this part 3 times because I keep getting burnt by these "when to pause" algorithms. 🤦🏻‍♂️